mmengine/evaluator/evaluator.py

```diff
@@ -123,10 +117,17 @@ def get_chunks(seq: Iterator, chunk_size=1):
                 if chunk:
                     yield chunk
 
+        if data is not None:
+            assert len(data_samples) == len(data), (
+                'outputs and data should have the same length, but got '
+                f'outputs length: {len(data_samples)} '
+                f'data length: {len(data)}')
+            data = get_chunks(iter(data), chunk_size)
+
         size = 0
         for output_chunk in get_chunks(iter(data_samples), chunk_size):
             if data is not None:
-                data_chunk = pseudo_collate(data[size:size + chunk_size])
+                data_chunk = pseudo_collate(next(data))  # type: ignore
```

Comment:
should the data here be chunked?

```suggestion
                data_chunk = pseudo_collate(get_chunks(iter(data), chunk_size))  # type: ignore
```

Reply:
the `data` here is a chunk generator